_drafts/release-workflows.md

```diff
@@ -0,0 +1,309 @@
+| `prerelease` | Version of the pre-release done from this branch. Must be formatted `<version>-<tag>`. The version is required, the `tag` is optional. Examples: `4.0.0-beta`, `4.0.0`. |
+
+**The order in which the branches are defines is meaningful, and it is used to determine the branch type and which release can be done from each branch.**
+
+##### Branches naming convention
+
+In a branch is defined as a `String` `semantic-release` will automatically determine the type of branch as follow:
+- If the branch name is a valid semver range (Only the format `1.x.x` or `1.x` are valid Git reference names) the branch will be considered a LTS branch, with its range based on the branch name, and release on the channel with the same name
+- The first branch in the `branches` `Array` after the LTS branches is considered the default branch (usually `master`).
+- Each branches in the `branches` `Array` after the default branch are considered future branches.
+
+For example:
+```json
+{
+  "release": {
+    "branches": [
+      "1.x.x",
+      "master",
+      "next",
+      {"branch": "alpha", "prerelease": true}
+    ]
+  }
+}
+```
+is equivalent to:
+```json
+{
+  "release": {
+    "branches": [
+      {"branch": "1.x.x", "range": "1.x.x", "channel": "latest"},
+      {"branch": "master", "channel": "latest"},
+      {"branch": "next", "channel": "next"},
+      {"branch": "alpha", "channel": "alpha", "preid": "alpha", "prerelease": true}
+    ]
+  }
+}
+```
+
+##### Configuration validation
+
+The `branches` configuration must follow these rules:
+- All branch name must be valid [Git reference](https://git-scm.com/docs/git-check-ref-format)
+- A branch cannot be named after a valid semver version (for example `1.0.0` or `v1.0.0` are invalid branch names) or a name that match the `tagFormat`.
+- A default branch is required.
+- LTS branches range must not overlap among themselves and the default one. For example `["1.x.x", 1.5.x]` is invalid.
+- LTS branches must be defined before the default branch.
+- Futures branches must be defined after the default branch.
+- A maximum of 2 future branch can be defined (one for features and one for breaking change)
+- Default and future branches cannot define a `range`.
+
+## Backward compatibility
+
+Non backward compatible changes:
+- Re-purpose the `branch` options as the current branch on which `semantic-release` is running
+- Pass the `channel` on which to release `nextRelease` to the `publish` plugins
+- publish plugins modifications
+  - If release already exists, make it available on `nextRelease.channel` (in case of github, remove the `prerelease` flag if the target channel is the default one)
+  - For the `npm` plugin, always publish on the dist-tag passed in `nextRelease.channel` or `latest` if `nextRelease.channel` is not set
+  - For the `github` plugin, set `prerelease` flag if the current branch is a future or prerelease
```

Comment:
maybe mention somewhere that CI-configuration needs to be adapted if semantic-release is configured to run only on `master` branch. Something I've set in many of my repositories

Reply:
Yes that would be in the migration guide